pb/proof.pb.go

```diff
@@ -30,13 +30,13 @@ type Proof struct {
 	End int64 `protobuf:"varint,2,opt,name=end,proto3" json:"end,omitempty"`
 	// Nodes hold the tree nodes necessary for the Merkle range proof.
 	Nodes [][]byte `protobuf:"bytes,3,rep,name=nodes,proto3" json:"nodes,omitempty"`
-	// LeafHash contains the namespace.ID if NMT does not have it and
+	// leaf_hash contains the namespace.ID if NMT does not have it and
 	// it should be proven. LeafHash is necessary to prove the Absence Proof.
 	// This field will be empty in case of Inclusion Proof.
-	LeafHash []byte `protobuf:"bytes,4,opt,name=leafHash,proto3" json:"leafHash,omitempty"`
-	// The isMaxNamespaceIgnored flag influences the calculation of the
+	LeafHash []byte `protobuf:"bytes,4,opt,name=leaf_hash,json=leafHash,proto3" json:"leaf_hash,omitempty"`
+	// The is_max_namespace_ignored flag influences the calculation of the
 	// namespace ID range for intermediate nodes in the tree.
-	IsMaxNamespaceIgnored bool `protobuf:"varint,5,opt,name=isMaxNamespaceIgnored,proto3" json:"isMaxNamespaceIgnored,omitempty"`
+	IsMaxNamespaceIgnored bool `protobuf:"varint,5,opt,name=is_max_namespace_ignored,json=isMaxNamespaceIgnored,proto3" json:"is_max_namespace_ignored,omitempty"`
```

Comment:
Based on https://github.com/celestiaorg/celestia-node/issues/2631#issuecomment-1697572961, I think the intention is for the JSON fields to be snake case. If that's the case, I don't expect this PR to resolve because:

```
json=leafHash
```

```
json=isMaxNamespaceIgnored
```

@distractedm1nd do you mind clarifying?

Reply:
Oops on re-review it looks like the `json:` is snake case but `json=` appears camel case. 

[not blocking] but what does the `json=` one do?

Reply:
@rootulp [Here](https://seb-nyberg.medium.com/customizing-protobuf-json-serialization-in-golang-6c58b5890356) there is an explanation.

Basically, `protobuf:"json=varName"` is used when you serialize struct by using the [protobuf json encoding](https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson) and `json:"name=var_name"` is used when you serialize it with [`encoding/json`](https://pkg.go.dev/encoding/json).

In [proto3 spec](https://protobuf.com/docs/language-spec#default-json-names), JSON names are converted from snake_case to lowerCamelCase